File: puzzles/ruby/stock-exchange.rb
@nb_values = gets.to_i
inputs = gets.split(" ")
max_value = -1
max_loss = 0

for i in 0..(@nb_values-1)
  value = inputs[i].to_i
  if value > max_value
    max_value = value
  else
    loss = max_value - value
    if loss > max_loss
      max_loss = loss
    end
  end
end

puts "#{-max_loss}"

File: puzzles/python3/stock_exchange.py
if __name__ == "__main__":
    max_value: int = 0
    max_loss: int = 0

    n: int = int(input())
    for v in input().split():
        value: int = int(v)
        if value > max_value:
            max_value = value
        elif value < max_value:
            loss: int = max_value - value
            if loss > max_loss:
                max_loss = loss

    print(-max_loss)
